webpack/scenes/HostCollections/Details/HostsTab/AddHostsModal.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
+import { useBulkSelect } from 'foremanReact/components/PF4/TableIndexPage/Table/TableHooks';
+import {
+  Modal,
+  ModalVariant,
+  Button,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import TableIndexPage from 'foremanReact/components/PF4/TableIndexPage/TableIndexPage';
+import SelectAllCheckbox from 'foremanReact/components/PF4/TableIndexPage/Table/SelectAllCheckbox';
+import { addHostsToCollection } from '../HostCollectionDetailsActions';
+
+const AddHostsModal = ({
```

Comment:
Tell your AI to look at webpack/components/extensions/HostDetails/Cards/HostCollectionsCard/HostCollectionsModal.js and see if there's a way to reuse that.